### ISSUE TITLE
p-accordion: click on p-accordion-header-text/icon doesn't toggle tab

### ISSUE THREAD

**Issue by DbkMG** [primefaces/primeng#15859]
### Describe the bug

A click on the p-accordion-header-text doesn't toggle the tab.

Just tried the upgrade from primeng 17.14.1 and tests are now failing because of this issue.

this is reproducable in primeng-showcase

### Environment

angular 17.3.10
primeng 17.18.1
tested in chrome & electron

### Angular version

17.3.10

### PrimeNG version

17.18.1

### Build / Runtime

Angular CLI App

### Language

TypeScript

### Node version (for AoT issues node --version)

20.11.1

### Steps to reproduce the behavior

1. go to https://primeng.org/accordion -> Basic Example
2. click on the header-text or icon of an accordion-tab (clicking on the tab-header itself works)
3. -> the tab won't open or close

### Expected behavior

an accordion-tab should also open/close by clicking on the header-text or icon. This worked in 17.14.1

**Comment by prx-lmo** [comment]
workaround till this gets released: 
downgrade to 17.17.0 and add in package.json
```
"overrides": {
    "primeng": {
      ".": "^17.17.0",
      "@angular/core": "^18.0.0",
      "@angular/common": "^18.0.0",
      "@angular/forms": "^18.0.0"
    }
  }
```

**Comment by rosenthalj** [comment]
This issue is a duplicate of Issue #15851

PR #15867  fixes this Issue #15859  and Issue #15851.
PR #15867 reverts PR #15838 which broke the Accordion Header functionality
